### ISSUE TITLE
indent -> CallExpression seems to be ignored.

### ISSUE THREAD

**Issue by gillez** [eslint/eslint#7687]
**Tell us about your environment**

ESLint v3.11.1
Node v6.9.1
npm v4.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
Default and VS Code ESLint

**Please show your full configuration:**

```
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "jquery": true
  },
  "globals": {
    "$": false,
    "angular": false,
    "listFunctions": false,
    "PouchDB": false,
    "yohelpers": false
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1,
        "CallExpression":{
          "arguments":"off"
        }
      }
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "off",
      "double"
    ],
    "semi": [
      "error",
      "always"
    ],
    "no-console": [
      // "warn"
      "off"
    ],
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_unused_"
      }

    ]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js

        if (errString !== "") {
          ngDialog.open({ template: "<p>" + errString + "</p>", 
              plain: true, className: 'ngdialog-theme-default'});
          return false;
        } else {

```

**What did you expect to happen?**

No warning about indentation off function parameters in call to ngDialog, as CallExpression is switched to "off".  This should be the default anyway.

**What actually happened? Please include the actual, raw output from ESLint.**

 86:15  error  Expected indentation of 12 spaces but found 14  indent

✖ 1 problem (1 error, 0 warnings)

*************
To get rid of this warning, I need to remove two spaces before "plain" so that it is indented by only two spaces (as is my normal code layout).  I would expect spaces within the function call to be ignored.  It doesn't matter what I set the CallExpression value to, I can't change this behaviour.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is actually working as intended. There is an object in your call expression, and ESLint is reporting it due to an incorrect indentation for objects. The fact that there happens to be a CallExpression around the object is not relevant to whether ESLint reports it.

For example, a similar error would be reported if you did this:

```js
var foo = { bar: 1,
                                          baz: 2});
```

> I would expect spaces within the function call to be ignored.

It wouldn't make sense for the rule to behave this way. If the rule ignored *all* indentation within function calls, regardless of how deeply nested it was, then code like this would get ignored:

```js
foo(function() {
  // All of the following code is inside the `foo` function call,
  // but the rule should still report it because the indentation is wrong.
                 var qux = 1;
        var bar = 2;
                           var baz = 3;
                if (blah)
{
                             qux();
            }
});
```

**Comment by gillez** [comment]
Thanks for the explanation.  
I understand the reasoning and it makes sense but is this new behaviour?  I ask because I have recently updated eslint and I've just started seeing these warnings all over the place.  The same codebase wasn't giving these warnings before.  Unfortunately I'm not sure what version I was using previously.

Thanks,
Giles

**Comment by kaicataldo** [comment]
I'm not sure when this changed, but it does sound it changed sometime between whatever version you were using and the latest.

Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
